### DISCUSSION TITLE
Question: Icon behavior?

### DISCUSSION THREAD

**Discussion by jmarshall9120** [kingyue737/vitify-admin#5] [Q&A]
I'm implementing this myself and thought you might be willing to share your wisdom, given your template works so well.  There is some sort of interesting interaction between Vuetify and Vite concerning Icons, that I haven't been able to find documented anywhere.  

This sort of thing doesn't work anymore after migrating to Vite
```
<v-col class="d-flex flex-shrink-1"
          ><v-btn @pointerdown="onLeftPointerDown"
            ><v-icon>mdi-arrow-left-bold</v-icon></v-btn
          ></v-col
        >
```
![image](https://user-images.githubusercontent.com/30184747/185807797-97ba1665-d6bb-4eef-9d01-ecbbbfb2b1e5.png)

I'm playing with your template, but I thought you might be willing to share your understanding here.  Its a pretty esoteric little bug.

**Comment by kingyue737** [comment]
Thanks for your interest in this template,  @jmarshall9120 !
Usage of icon in this template is documented [here](https://kingyue737.github.io/vitify-docs/guide/icon.html). I choose `@mdi/js` instead of `@mdi/font` in order not to include all the icons in the bundle which significantly increases the bundle size .
```sh
pnpm install -D @mdi/js
```

```ts
new Vuetify ({
  /* ... */
  icons: {
    iconfont: 'mdiSvg',     // not 'mdi' here
      values: {
        ...icons,
      },
    }
  }
```

But the con is that icons need to be registered manually.

You need to import `mdiArrowLeftBold` in [`@/plugins/icon.ts`](https://github.com/kingyue737/vitify-admin/blob/main/src/plugins/icons.ts)

```ts
// src/plugins/icons.ts
import {
  mdiAlert,
  mdiEye,
  mdiEyeOff,
  mdiBell,
  /* other icons ... */
  mdiArrowLeftBold,
} from '@mdi/js'

// icons in `mdi` object will be registered in Vuetify
const mdi = {
  mdiAlert,
  mdiEye,
  mdiEyeOff,
  mdiBell,
/* other icons ... */
  mdiArrowLeftBold
}
```
Then use `mdiArrowLeftBold` in vue template:
```html
<v-icon>$mdi-arrow-left-bold</v-icon>
```
Don't forget the `$` in front of icon name!

You can also switch back to `@mdi/font` if you don't care about bundle size or don't want to manually import icons. Ref to [Vuetify official docs](https://vuetifyjs.com/en/features/icon-fonts/)

**Comment by kingyue737** [comment]
If you are using `@mdi/font`, don't forget to import mdi css somewhere in your code

```ts
import '@mdi/font/css/materialdesignicons.css'
```
